### PR TITLE
[12.0][ADD] helpdesk_mgmt_ticket_assignment_massive

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Adds an action to assign status and assigned user in bulk.

### CODE REVIEW

File: helpdesk_mgmt_ticket_assignment_massive/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Helpdesk Ticket Assignment Massive",
+    "summary": """
+        Adds an action to assign status and assigned user in bulk.
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_assignment_massive/__manifest__.py]: Suggested:
```suggestion
        Adds an action for tickets to assign status and assigned user in bulk.
```

File: helpdesk_mgmt_ticket_assignment_massive/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "version": "12.0.1.0.0",
+    "license": "LGPL-3",
+    "category": "After-Sales",
+    "author": "Solvos",
+    "website": "https://github.com/solvosci/slv-helpdesk",
+    "depends": [
+        "helpdesk_mgmt",
+    ],
+    "data": [
+        "wizards/assignement_massive.xml",
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_assignment_massive/__manifest__.py]: Typo:
```suggestion
        "wizards/assignment_massive.xml",
```

File: helpdesk_mgmt_ticket_assignment_massive/wizards/assignement_massive.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                    <field name="ticket_ids"/>
+                </group>
+                <footer>
+                    <button name="assigned_user" string="Submit" type="object" class="btn-primary"/>
+                    <button string="Cancel" special="cancel" class="btn-secondary"/>
+                </footer>
+            </form>   
+        </field>
+    </record>
+</odoo>
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_assignment_massive/wizards/assignement_massive.xml]: Missing EOF line

File: helpdesk_mgmt_ticket_assignment_massive/wizards/assignement_massive.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    stage_id = fields.Many2one('helpdesk.ticket.stage', string="Stage")
+
+    @api.depends('ticket_ids, stage_id')
+    def assigned_user(self):
+        for ticket in self.ticket_ids:
+            if(self.user_id):
+                ticket.write({
+                    'user_id': self.user_id.id
+                })
+            if(self.stage_id):
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_assignment_massive/wizards/assignement_massive.py]: This code doesn't cover two corner cases:
- If `user_id` is not filled, the tickets aren't modified. But, if we want to unassign those tickets?
- If any ticket has a `team_id` incompatible with the selected `user_id`, an error should be fired and process stopped.

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments

**Comment by ChristianSantamaria** [comment]
> If user_id is not filled, the tickets aren't modified. But, if we want to unassign those tickets?
What would be the best option for this create a checkbox indicating if you want to deallocate the user or create another lower button to deallocate?

**Comment by dalonsod** [comment]
@ChristianSantamaria I'll test it as soon as possible

**Comment by dalonsod** [comment]
>What would be the best option for this create a checkbox indicating if you want to deallocate the user or create another lower button to deallocate?

Other options is the selected (leaving empty the field does the trick) and adding a legend in the wizard form explaining that, like "Please select the new assigned user. If it's empty, the selected tickets will become unassigned"

**Comment by dalonsod** [comment]
Locally tested, faced these issues:

- When a user is not valid, the message doesn't show affected ticket number, it should be useful IMO.
- When the user is not filled, testing with three random tickets the following error message was fired: "The assigned user False is not associated with the team Consultants ". When the user is not filled, no checks are needed.
- The ticket list in the wizard form has the label "Ticket" (better "Selected tickets"). Since a custom tree view is not defined, a too long column list is shown. Consider using `widget="many2many_tags"` in this case.

**Comment by dalonsod** [comment]
@ChristianSantamaria please test the last modifications made. I think only one condition should be tested in this case:
- If user is filled, check if user belongs to current ticket team, if assigned.
- If there's no team for the ticket, every user could be assigned to the ticket.
- Every ticket can be user unassigned.

Tested in a database created from scratch, and only installing this addon.

File: helpdesk_mgmt_ticket_assignment_massive/wizards/assignment_massive.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        for ticket in self.ticket_ids:
+            if self.user_id:
+                if ticket.team_id and \
+                   self.user_id.id not in ticket.team_id.user_ids.ids:
+                    raise ValidationError(_(
+                        'In ticket %s the assigned user %s '
+                        'is not associated with team %s'
+                        % (ticket.number, self.user_id.name,
+                           ticket.team_id.name)
+                    ))
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_assignment_massive/wizards/assignment_massive.py]: Wrong string injection, it actually alters origin string:
```suggestion
                    raise ValidationError(_(
                        'In ticket %s the assigned user %s '
                        'is not associated with team %s'                        
                    ) % (ticket.number, self.user_id.name,
                           ticket.team_id.name)
                    )
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
Locally tested, LGTM, only the issue describe on other comment

**Review by dalonsod — APPROVED** [review]
LGTM

@lmiguens-solvos could you review? Thanks!

**Review by lmiguens-solvos — APPROVED** [review]
LGTM

**Comment by dalonsod** [comment]
Let's merge!

**Pull request merged**